### PR TITLE
feat: initialize project layout

### PR DESCRIPTION
**Pull request by Hidayathamir**
I initialize project layout, combining [brix](https://github.com/brix-go/brix), [golang-standards](https://github.com/golang-standards/project-layout) and our current folder structure in eraspace.

## How to Run

1. Rename `.env-example` to `.env`.

```shell
mv .env-example .env
```

2. Run database using docker compose in folder `deployment`.

```shell
cd deployment && sudo docker compose up
```

3. Run `main.go` in other terminal.

```shell
go run main.go
```

4. Test dummy API.
```shell
curl --location 'http://localhost:4000/api/v1/baz/true'
```
```shell
curl --location 'http://localhost:4000/api/v1/baz/false'
```

### CODE REVIEW

File: internal/api/v1/cart/service/service.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"fmt"
+
+	"github.com/gin-gonic/gin"
+	"github.com/spacetronot-research-team/cart-service/internal/api/v1/cart/repository"
+	"github.com/spacetronot-research-team/cart-service/internal/pkg/carterr"
+	"github.com/spacetronot-research-team/cart-service/pkg/eraerr"
+)
+
+type ICartService interface {
+	Baz(ctx *gin.Context, isShouldError string) (string, error)
```
Comment (irvankadhafi) [inline internal/api/v1/cart/service/service.go:14]: menurutku kalo udah di layer service pake context.Context aja.
solnya *gin.Context bisa di masukin ke function dg parameter bertype context.Context

`Baz(ctx context.Context, isShouldError string) (string, error)`

File: internal/api/v1/cart/service/service.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	"errors"
+	"fmt"
+
+	"github.com/gin-gonic/gin"
+	"github.com/spacetronot-research-team/cart-service/internal/api/v1/cart/repository"
+	"github.com/spacetronot-research-team/cart-service/internal/pkg/carterr"
+	"github.com/spacetronot-research-team/cart-service/pkg/eraerr"
+)
+
+type ICartService interface {
```
Comment (irvankadhafi) [inline internal/api/v1/cart/service/service.go:13]: ini naming perlu ada convention juga dikita, 
pake awalan I atau cukup CartService

**Comment by Hidayathamir** [comment]
move to here https://github.com/spacetronot-research-team/cart-service/pull/2

**Comment by irvankadhafi** [comment]
test

**Review by irvankadhafi — COMMENTED** [review]
check ya

**Pull request closed without merging**